### PR TITLE
docs: document file size configuration for deploy endpoint

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR adds an explanation on how to properly configure Camunda for Resource Deployment REST endpoint.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

closes: #4442

### CODE REVIEW

**Comment by nicpuppa** [comment]
I'm not sure which labels add here 👀

File: docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md [hunk 1]
```diff
@@ -24,6 +24,18 @@ For Self-Managed, the host and port depend on your configuration. The context pa
 
 See [the interactive Camunda 8 REST API Explorer][camunda-api-explorer] for specifications, example requests and responses, and code samples of interacting with the Camunda 8 REST API.
 
+## Deployment API
+
+In Zeebe you can configure `maxMessageSize` in the [Gateway](../../self-managed/zeebe-deployment/configuration/gateway.md#zeebegatewaynetwork) and in the [Broker](../../self-managed/zeebe-deployment/configuration/broker.md#zeebebrokernetwork) configuration. The default value is 4MB.
```
Comment (tmetzke) [inline docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md:29]:
```suggestion
You can define a `maxMessageSize` in the [Gateway](../../self-managed/zeebe-deployment/configuration/gateway.md#zeebegatewaynetwork) and [Broker](../../self-managed/zeebe-deployment/configuration/broker.md#zeebebrokernetwork) configuration. The default value is 4MB.
```

File: docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md [hunk 2]
```diff
@@ -24,6 +24,18 @@ For Self-Managed, the host and port depend on your configuration. The context pa
...
+To allow users to deploy resources via REST endpoint we added 2 Spring configuration properties, by default the value is 4MB:
```
Comment (tmetzke) [inline docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md:30]:
```suggestion
Accordingly, the [Deploy resources](/apis-tools/camunda-api-rest/specifications/deploy-resources.mdx) REST endpoint by default allows uploading single files and overall data of up to 4MB. You can adjust this configuration via the following properties:
```

File: docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md [hunk 3]
```diff
@@ -24,6 +24,18 @@ For Self-Managed, the host and port depend on your configuration. The context pa
...
+
+```properties
+spring.servlet.multipart.max-file-size=4MB
+spring.servlet.multipart.max-request-size=4MB
+```
+
+If you are using an higher value for `maxMessageSize` you need to increase the value of the above properties to match the `maxMessageSize` value.
```
Comment (tmetzke) [inline docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md:37]:
```suggestion
If you define a different `maxMessageSize`, we recommend configuring the values of the above properties accordingly.
```

File: versioned_docs/version-8.6/apis-tools/camunda-api-rest/camunda-api-rest-overview.md [hunk 4]
```diff
@@ -24,6 +24,18 @@ For Self-Managed, the host and port depend on your configuration. The context pa
 
 See [the interactive Camunda 8 REST API Explorer][camunda-api-explorer] for specifications, example requests and responses, and code samples of interacting with the Camunda 8 REST API.
 
+## Deployment API
+
+In Zeebe you can configure `maxMessageSize` in the [Gateway](../../self-managed/zeebe-deployment/configuration/gateway.md#zeebegatewaynetwork) and in the [Broker](../../self-managed/zeebe-deployment/configuration/broker.md#zeebebrokernetwork) configuration. The default value is 4MB.
```
Comment (tmetzke) [inline versioned_docs/version-8.6/apis-tools/camunda-api-rest/camunda-api-rest-overview.md:29]:
```suggestion
You can define a `maxMessageSize` in the [Gateway](../../self-managed/zeebe-deployment/configuration/gateway.md#zeebegatewaynetwork) and [Broker](../../self-managed/zeebe-deployment/configuration/broker.md#zeebebrokernetwork) configuration. The default value is 4MB.
```

**Review by tmetzke — APPROVED** [review]
👍 Good spot, good content, @nicpuppa 🦖 
🔧 I added some wording and grammar suggestions, nothing big. Pre-approving content-wise already.

File: docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md [hunk 5]
```diff
@@ -24,6 +24,18 @@ For Self-Managed, the host and port depend on your configuration. The context pa
 
 See [the interactive Camunda 8 REST API Explorer][camunda-api-explorer] for specifications, example requests and responses, and code samples of interacting with the Camunda 8 REST API.
 
+## Deployment API
```
Comment (tmetzke) [inline docs/apis-tools/camunda-api-rest/camunda-api-rest-overview.md:27]:
```suggestion
### Deployment API
```

File: versioned_docs/version-8.6/apis-tools/camunda-api-rest/camunda-api-rest-overview.md [hunk 6]
```diff
@@ -24,6 +24,18 @@ For Self-Managed, the host and port depend on your configuration. The context pa
 
 See [the interactive Camunda 8 REST API Explorer][camunda-api-explorer] for specifications, example requests and responses, and code samples of interacting with the Camunda 8 REST API.
 
+## Deployment API
```
Comment (tmetzke) [inline versioned_docs/version-8.6/apis-tools/camunda-api-rest/camunda-api-rest-overview.md:27]:
```suggestion
### Deployment API
```

**Comment by mesellings** [comment]
@nicpuppa I've committed some edits directly to the branch - reworded/restructured this section for clarity. I haven't applied this to the 8.6 file, but if you're happy with these edits I can do that as well if you like and then approve?

**Comment by nicpuppa** [comment]
@mesellings go ahead 👍

**Comment by mesellings** [comment]
Done - backported to 8.6.

**Review by mesellings — APPROVED** [review]
Lgtm 👍

**Pull request merged**